### PR TITLE
Some partest improvements

### PR DESCRIPTION
**Pull request by szeiger**
- Reverse diff direction (check -> log)
- Always show logs and diffs when running only individual tests
- Simplify some internals

### CODE REVIEW

File: src/partest/scala/tools/partest/nest/AbstractRunner.scala [hunk 1]
```diff
@@ -8,25 +8,32 @@ package partest
...
+import java.util.concurrent.TimeUnit.NANOSECONDS
 
-abstract class AbstractRunner {
+class AbstractRunner(val config: RunnerSpec.Config, protected final val testSourcePath: String, val fileManager: FileManager) {
 
-  val config: RunnerSpec.Config
-
-  lazy val nestUI: NestUI = new NestUI(
+  // lazy because colorEnabled is overridden in SBTRunner
+  // TODO: integrate NestUI into AbstractRunner
```
Comment (szeiger) [inline src/partest/scala/tools/partest/nest/AbstractRunner.scala:24]: I was tempted to continue down this path but I should probably spend my time elsewhere at the moment.

File: src/partest/scala/tools/partest/nest/Runner.scala [hunk 2]
```diff
@@ -758,128 +758,16 @@ object Properties extends scala.util.PropertiesTrait {
...
   val scalacOpts: String = PartestDefaults.scalacOpts) {
 
-  import PartestDefaults.{ numThreads, waitTime }
-
-  setUncaughtHandler
-
-  // TODO: make this immutable
-  PathSettings.testSourcePath = testSourcePath
-
-  val durations = collection.concurrent.TrieMap[File, Long]()
```
Comment (som-snytt) [inline src/partest/scala/tools/partest/nest/Runner.scala:780]: Does no one report on longest-running tests? or is that in sbt?

File: src/partest/scala/tools/partest/nest/ConsoleRunner.scala [hunk 3]
```diff
@@ -7,13 +7,8 @@ package scala.tools.partest
 package nest
 
 class ConsoleRunner(val config: RunnerSpec.Config) extends AbstractRunner {
-  val suiteRunner = new SuiteRunner (
-    testSourcePath = config.optSourcePath getOrElse PartestDefaults.sourcePath,
-    fileManager = new FileManager(ClassPath split PathResolver.Environment.javaUserClassPath map (Path(_))), // the script sets up our classpath for us via ant
-    updateCheck = config.optUpdateCheck,
-    failed = config.optFailed,
-    noexec = config.optNoExec,
```
Comment (som-snytt) [inline src/partest/scala/tools/partest/nest/ConsoleRunner.scala:15]: The most painful part of adding noexec.

**Review by som-snytt — APPROVED** [review]
I liked what I saw. Woe to anyone who resurrects ant runner or dreams of gradle runner. My only question was about reporting the heaviest tests.

**Review by dwijnand — APPROVED** [review]
SGTM!

**Comment by adriaanm** [comment]
I've been quietly suffering through the reverse diff direction ever since I introduced the bug (as penance for introducing it?). I feel much lighter! 😅

**Comment by som-snytt** [comment]
We suffer together as a community. Off to go rebase now!

File: src/partest/scala/tools/partest/nest/Runner.scala (hunk 2 above)
Comment (szeiger) [reply src/partest/scala/tools/partest/nest/Runner.scala:780]: I've seen some individual tests report durations (if it's over a set threshold I assume) but apparently we don't use these collected durations anywhere.

**Pull request merged**